File: solutions/_217_contains_duplicate/solution.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        size_t last = nums.size() - 1;

        for (size_t i = 0; i < last; ++i)
        {
            if (nums[i] == nums[i + 1])
                return true;
        }

        return false;
    }
};

/*
https://leetcode.com/problems/contains-duplicate/
*/

File: solutions/_217_contains_duplicate/solution.php
<?php
class Solution
{
    /**
     * @param int[] $nums
     * @return bool
     */
    function containsDuplicate(array $nums): bool
    {
        $seen = [];
        foreach ($nums as $num) {
            if (isset($seen[$num])) {
                return true;
            }
            $seen[$num] = true;
        }
        return false;
    }
}
